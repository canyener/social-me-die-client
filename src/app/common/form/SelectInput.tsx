import { FieldRenderProps } from 'react-final-form'
import { Form, FormFieldProps, Label, Select } from 'semantic-ui-react'

interface IProps 
  extends FieldRenderProps<string, HTMLSelectElement>, 
    FormFieldProps { }

const SelectInput: React.FC<IProps> = ({
  input,
  width,
  options,
  placeholder,
  meta: { touched, error }
}) => {
  return (
    <Form.Field error={touched && !!error} width={width}>
      <Select 
        value={input.value}
        onChange={(_e, data) => input.onChange(data.value)}
        placeholder={placeholder}
        options={options}
      />
      {
        touched && error && (
          <Label basic color='red'>{error}</Label>
        )
      }
    </Form.Field>
  )
}

export default SelectInput