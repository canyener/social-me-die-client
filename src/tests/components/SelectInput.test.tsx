import { shallow } from 'enzyme'

import SelectInput from '../../app/common/form/SelectInput'

import { fakeCategories } from '../fixtures/seed'

test('Should render SelectInput correctly without any given errors', () => {

  const wrapper = shallow(<SelectInput 
    width='11'
    placeholder='test select input placeholder'
    options={fakeCategories}
    input={{
      name:'TestSelectInputName',
      value:'TestSelectInputValue',
      onBlur: jest.fn(),
      onChange: jest.fn(),
      onFocus: jest.fn()
    }}
    meta= {{
      touched: false,
      error: undefined
    }}
  />)

  expect(wrapper).toMatchSnapshot()
})

test('Should set Form.Field error to true and render Label with given error message', () => {
  const wrapper = shallow(<SelectInput 
    width='11'
    placeholder='test select input placeholder'
    options={fakeCategories}
    input={{
      name:'TestSelectInputName',
      value:'TestSelectInputValue',
      onBlur: jest.fn(),
      onChange: jest.fn(),
      onFocus: jest.fn()
    }}
    meta= {{
      touched: true,
      error: 'test select input error message'
    }}
  />)

  expect(wrapper).toMatchSnapshot()
})