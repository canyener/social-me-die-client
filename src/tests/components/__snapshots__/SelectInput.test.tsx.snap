// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render SelectInput correctly without any given errors 1`] = `
<FormField
  error={false}
  width="11"
>
  <Select
    onChange={[Function]}
    options={
      Array [
        Object {
          "key": "fakeCategoryOption1",
          "text": "Fake Category Option 1",
          "value": "fakeCategoryOption1",
        },
        Object {
          "key": "fakeCategoryOption2",
          "text": "Fake Category Option 2",
          "value": "fakeCategoryOption2",
        },
        Object {
          "key": "fakeCategoryOption3",
          "text": "Fake Category Option 3",
          "value": "fakeCategoryOption3",
        },
      ]
    }
    placeholder="test select input placeholder"
    value="TestSelectInputValue"
  />
</FormField>
`;

exports[`Should set Form.Field error to true and render Label with given error message 1`] = `
<FormField
  error={true}
  width="11"
>
  <Select
    onChange={[Function]}
    options={
      Array [
        Object {
          "key": "fakeCategoryOption1",
          "text": "Fake Category Option 1",
          "value": "fakeCategoryOption1",
        },
        Object {
          "key": "fakeCategoryOption2",
          "text": "Fake Category Option 2",
          "value": "fakeCategoryOption2",
        },
        Object {
          "key": "fakeCategoryOption3",
          "text": "Fake Category Option 3",
          "value": "fakeCategoryOption3",
        },
      ]
    }
    placeholder="test select input placeholder"
    value="TestSelectInputValue"
  />
  <Label
    basic={true}
    color="red"
  >
    test select input error message
  </Label>
</FormField>
`;
